ppbtree/ppbtree.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+"""
+based on Clement Michard (c) 2015
+Vladyslav Khardel (c) 2017
+"""
+
+class Node:
+    def __init__(self, value=None):
+        self.value = value
+        self.left = None
+        self.right = None
+
+
+def add(node, value):
+    if node.value is None:
+        node.value = value
+    else:
+        if value <= node.value:
+            if node.left is None:
+                node.left = Node(value)
+            else:
+                add(node.left, value)
+        else:
+            if node.right is None:
+                node.right = Node(value)
+            else:
+                add(node.right, value)
+
+def print_tree(current_node, nameattr='value', left_child='left', right_child='right', indent='', last='updown'):
+
+    if hasattr(current_node, nameattr):
+        name = lambda node: getattr(node, nameattr)
+    else:
+        name = lambda node: str(node)
+
+    up = getattr(current_node, left_child)
```

Comment:
Small nit-pick: personally I favor these reversed `up=right` + `down=left`. Easier for me to visualize if you imagine rotating the tree 90 degrees to the right. But seems a matter of opinion